File: Aula11/Exercicio/Python/exercicio.py
class Produto:
    def __init__(self, nome, preco, quantidade, categoria):
        self.__nome = nome
        self.__preco = preco
        self.__quantidade = quantidade
        self.__categoria = categoria

    def aplicar_desconto(self, percentual):
        self.__preco -= self.__preco * (percentual / 100)
        return f"Novo preço de '{self.__nome}': R$ {self.__preco:.2f}"

    def atualizar_estoque(self, qtd):
        self.__quantidade += qtd
        return f"Estoque de '{self.__nome}': {self.__quantidade} unidades"

    def exibir_detalhes(self):
        return (f"Produto: {self.__nome} | Categoria: {self.__categoria} "
                f"| Preço: R$ {self.__preco:.2f} | Qtd: {self.__quantidade}")

    def verificar_disponibilidade(self):
        return "Disponível" if self.__quantidade > 0 else "Indisponível"

# Instanciando 3 objetos
p1 = Produto("Notebook", 3500.00, 10, "Eletrônicos")
p2 = Produto("Cadeira Gamer", 1200.00, 5, "Móveis")
p3 = Produto("Mouse", 150.00, 0, "Periféricos")

print(p1.exibir_detalhes())
print(p2.aplicar_desconto(10))
print(p3.verificar_disponibilidade())

File: Aula11/Java/exercicio.java
public class Produto {
    private String nome;
    private double preco;
    private int quantidade;
    private String categoria;

    public Produto(String nome, double preco, int quantidade, String categoria) {
        this.nome = nome;
        this.preco = preco;
        this.quantidade = quantidade;
        this.categoria = categoria;
    }

    public String aplicarDesconto(double percentual) {
        this.preco -= this.preco * (percentual / 100);
        return "Novo preço de '" + nome + "': R$ " + String.format("%.2f", preco);
    }

    public String atualizarEstoque(int qtd) {
        this.quantidade += qtd;
        return "Estoque de '" + nome + "': " + quantidade + " unidades";
    }

    public String exibirDetalhes() {
        return "Produto: " + nome + " | Categoria: " + categoria
             + " | Preço: R$ " + String.format("%.2f", preco)
             + " | Qtd: " + quantidade;
    }

    public String verificarDisponibilidade() {
        return quantidade > 0 ? "Disponível" : "Indisponível";
    }

    public static void main(String[] args) {
        Produto p1 = new Produto("Notebook", 3500.00, 10, "Eletrônicos");
        Produto p2 = new Produto("Cadeira Gamer", 1200.00, 5, "Móveis");
        Produto p3 = new Produto("Mouse", 150.00, 0, "Periféricos");

        System.out.println(p1.exibirDetalhes());
        System.out.println(p2.aplicarDesconto(10));
        System.out.println(p3.verificarDisponibilidade());
    }
}
